src/app/campaign/availability/availability.component.ts

```diff
@@ -1,59 +1,97 @@
     <h2>Impressions</h2>
-    <p *ngIf="flight && (!flight.startAt || !flight.endAt || !flight.set_inventory_uri || !(flight.zones && flight.zones.length))">
+    <p *ngIf="cantShowInventory(); else inventory">
       Please select Start and End Dates, Series, and Zones to view inventory.
     </p>
-    <section *ngFor="let zone of availabilityZones">
-      <h3>{{ getZoneName(zone.zone) }}</h3>
-      <div class="row head">
-        <div class="date">Week</div>
-        <div class="avail">Available</div>
-        <div class="goal">Goal</div>
-        <div class="edit"><prx-icon name="lock" size="14px" color="secondary"></prx-icon></div>
-      </div>
-      <ng-container *ngFor="let week of zone.totals.groups">
-        <div class="row">
-          <div class="date">
-            <button class="btn-link"
-              (click)="toggleZoneWeekExpanded(zone.zone, week.startDate)"
-              (mouseover)="toggleZoneWeekHover(zone.zone, week.startDate)"
-              (mouseleave)="toggleZoneWeekHover(zone.zone, week.startDate)">
-              <prx-icon
-                [class.hide]="!(zoneWeekHover[zone.zone + '-' + week.startDate] || zoneWeekExpanded[zone.zone + '-' + week.startDate])"
-                name="arrows-alt-v" size="6px" color="primary">
-              </prx-icon>
-              {{ week.startDate | date:'M/dd' }}
-            </button>
-          </div>
-          <div class="avail">{{ (week.allocated + week.availability) | largeNumber }}</div>
-          <div class="goal"></div>
-          <div class="edit">
-            <button class="btn-link" aria-label="Edit">
-              <prx-icon name="pencil" size="14px" color="primary"></prx-icon>
-            </button>
-          </div>
+    <ng-template #inventory>
+      <grove-goal-form [flight]="flight" (goalChange)="goalChange.emit($event)"></grove-goal-form>
+      <p class="error" *ngIf="allocationPreviewError">Got error #{{ allocationPreviewError.status }} from allocation preview</p>
+      <section *ngFor="let zone of availabilityZones">
+        <h3>{{ getZoneName(zone.zone) }}</h3>
+        <div class="row head">
+          <div class="date">Week</div>
+          <div class="avail">Available</div>
+          <div class="goal">Goal</div>
+          <div class="edit"><prx-icon name="lock" size="14px" color="secondary"></prx-icon></div>
         </div>
-        <ng-container *ngIf="zoneWeekExpanded[zone.zone + '-' + week.startDate]">
-          <div class="row expand" *ngFor="let day of week.groups">
-            <div class="date"><span>&mdash;</span><span>{{ day.startDate | date:'M/dd' }}</span></div>
-            <div class="avail"><span>&mdash;</span><span>{{ (day.allocated + day.availability) | largeNumber }}</span></div>
-            <div class="goal"></div>
-            <div class="edit"></div>
+        <ng-container *ngFor="let week of zone.totals.groups">
+          <div class="row" [class.row-highlight]="week.allocated + week.availability < week.allocationPreview">
```

Comment:
I'm not sure I understand this row-highlight correctly. I added here what I thought I saw in designs, but it doesn't quite make sense for this. It seems according to this design I should highlight a row if the availability is less than the allocation preview, but I'm not sure why the system would return an allocation preview lower than is available. 
<img width="265" alt="Screen Shot 2019-10-24 at 4 45 21 PM" src="https://user-images.githubusercontent.com/2250413/67527922-ba646f80-f67d-11e9-93ec-7c6085b01699.png">

The criteria in the ticket is this, but Augury already seems to handle these cases because I get an error if daily_min cannot be met instead of a partial result. Unfortunately those error messages don't current come through the hal remote, but we could sort it out.
* Validation errors, if Augury couldn't meet your total/daily_min goals:
  * Highlight offending Total/Daily-Min field in red
  * Change table text to red for offending day/total rows

Reply:
Sounds like a question for @brandonhundt

Reply:
#68 has been added to address validation